src/context/importantMessage.tsx

```diff
@@ -0,0 +1,39 @@
+import React, {
+  createContext,
+  FunctionComponent,
+  useState,
+  Dispatch,
+  SetStateAction
+} from "react";
+
+type MessageContent = {
+  title: string;
+  description?: string;
+  featherIconName?: string;
+  action?: {
+    callback: () => void;
+    label: string;
+  };
+};
+
+export const ImportantMessageContentContext = createContext<MessageContent | null>(
+  null
+);
+export const ImportantMessageSetterContext = createContext<
+  Dispatch<SetStateAction<MessageContent | null>>
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+>(() => {});
+
+export const ImportantMessageContextProvider: FunctionComponent = ({
+  children
+}) => {
+  const [message, setMessage] = useState<MessageContent | null>(null);
```

Comment:
would you want to consider making this a Banner component so that the code for the message context is not coupled with the Banner's props. 

Of course that is for the case of one single banner.. .

Reply:
Hmm, yea I'll look into this in another PR where I'll handle multiple messages better.